apollo-router/src/plugins/telemetry/tracing/apollo_telemetry.rs

```diff
@@ -316,457 +312,8 @@ impl From<&StudioGraph> for ReporterGraph {
-
-        let key = stats_report_key(&op_name, query);
-        assert_eq!(expected, key);
-    }
-
-    #[test]
-    fn it_handles_query_specified_name() {
-        let expected =
-            "# OneProduct\nquery OneProduct($limit: Int!) { products(limit: $limit) { upc, name, price } }";
-        let op_name = Value::String("".into());
-        let query = "query OneProduct($limit: Int!) {\n  products(limit: $limit) {\n    upc,\n    name,\n    price\n  }\n}";
-
-        let key = stats_report_key(&op_name, query);
-        assert_eq!(expected, key);
-    }
-
-    #[test]
-    fn it_handles_same_out_of_band_and_query_specified_name() {
-        let expected =
-            "# OneProduct\nquery OneProduct($limit: Int!) { products(limit: $limit) { upc, name, price } }";
-        let op_name = Value::String("OneProduct".into());
-        let query = "query OneProduct($limit: Int!) {\n  products(limit: $limit) {\n    upc,\n    name,\n    price\n  }\n}";
-
-        let key = stats_report_key(&op_name, query);
-        assert_eq!(expected, key);
-    }
-
-    #[test]
-    fn it_handles_same_out_of_band_and_query_specified_name_multiple_queries() {
-        let expected = "# OneProduct\nquery OneProduct { __typename } ";
-        let op_name = Value::String("OneProduct".into());
-        let query = "query OneProduct { __typename } query AnotherProduct { __typename }";
-
-        let key = stats_report_key(&op_name, query);
-        assert_eq!(expected, key);
-    }
-
-    #[test]
-    fn it_handles_missing_out_of_band_and_query_specified_name_multiple_queries() {
-        let expected = GRAPHQL_UNKNOWN_OPERATION_NAME;
-        let op_name = Value::String("YetAnotherProduct".into());
-        let query = "query OneProduct { __typename } query AnotherProduct { __typename }";
-
-        let key = stats_report_key(&op_name, query);
-        assert_eq!(expected, key);
-    }
-
-    #[test]
-    fn it_handles_no_out_of_band_name_and_multiple_queries() {
-        let expected = GRAPHQL_UNKNOWN_OPERATION_NAME;
-        let op_name = Value::String("".into());
-        let query = "query OneProduct { __typename } query AnotherProduct { __typename }";
-
-        let key = stats_report_key(&op_name, query);
-        assert_eq!(expected, key);
-    }
-
-    //TODO: This test won't work because we aren't doing any validation. I'm leaving
-    //the test to remember that at some point it will need to be addressed. Perhaps initially
-    //in the router-bridge enhancements.
-    #[test]
-    #[ignore]
-    fn it_handles_invalid_out_of_band_name() {
-        let expected = GRAPHQL_VALIDATION_FAILURE;
-        let op_name = Value::String("anythingo r missing".into());
-        let query = "query OneProduct { __typename } query AnotherProduct { __typename }";
-
-        let key = stats_report_key(&op_name, query);
-        assert_eq!(expected, key);
+    async fn export(&mut self, _batch: Vec<SpanData>) -> ExportResult {
```

Comment:
In general: I'm nervous that we've moved the stats reporting stack from an opentelemetry base which was certainly not ideal, but was at least written to scale, to storing the data in something relatively unproven (dashmap).

I'm not saying this is definitely wrong, just observing that we'll really have to be sure that performance/scalability/reliability is not impacted.

Reply:
I agree this is not ideal, but it's the only option we have right now as the histogram generation provided by Otel will be incompatible with our custom implementation and we have to move away from spans to avoid sampling.

In terms of performance I tested locally and there seems to be no difference in performance.